app/controllers/AccountController.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import connectors.AuthConnector
+import model.Forms._
+import model.{GatekeeperSessionKeys, LoginDetails}
+import play.api.Logger
+import play.api.i18n.Messages
+import play.api.mvc._
+import uk.gov.hmrc.play.frontend.controller.FrontendController
+import uk.gov.hmrc.play.http.SessionKeys
+import utils.{GatekeeperAuthProvider, GatekeeperAuthWrapper}
+import views.html.login._
+
+import scala.concurrent.Future
+
+object AccountController extends AccountController {
+  val authConnector = AuthConnector
+
+  override def authProvider = GatekeeperAuthProvider
+}
+
+trait AccountController extends FrontendController with GatekeeperAuthWrapper {
+
+  val authConnector: AuthConnector
+
+  val welcomePage = routes.DashboardController.dashboardPage
+
+  val loginPage: Action[AnyContent] = redirectIfLoggedIn(welcomePage) {
+    implicit request => implicit hc => Future.successful(Ok(login(loginForm)))
+  }
+
+  val authenticateAction = Action.async { implicit request =>
+    loginForm.bindFromRequest().fold(
+      errors => Future.successful(BadRequest(login(loginForm))),
+      loginDetails => processLogin(loginDetails)
+    )
+  }
+
+  def logoutAction = Action(implicit request => Redirect(routes.AccountController.loginPage()).removingFromSession(SessionKeys.authToken))
```

Comment:
The nittiest of picks - and feel free to ignore it if you like.  I would prefer this to be spread across three lines. 

This reads more easily to me: 
```scala
  def logoutAction = Action { implicit request =>
     Redirect(routes.AccountController.loginPage()).removingFromSession(SessionKeys.authToken)
  }
```